application/src/main/java/it/smartcommunitylabdhub/core/config/SecurityConfig.java

```diff
@@ -144,6 +165,22 @@ private JwtDecoder jwtDecoder() {
         return jwtDecoder;
     }
 
+    private JwtDecoder exchangeJwtDecoder() throws JOSEException {
+        JWK jwk = keyStoreConfig.getJWKSetKeyStore().getJwk();
```

Comment:
1. autowire kestore as dep and *NOT* config.
2. you need to check for null both on keystore and jwk